### PR TITLE
remove depreacted binding-generator

### PR DESCRIPTION
**Pull request by novain1102**
Re: #https://github.com/cocos/cocos-engine/issues/12859

### Changelog

*

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: native/tools/swig-config/assets.i [hunk 1]
```diff
@@ -77,12 +77,14 @@
...
+%rename(getHash) cc::Material::getHashForMaterial;
 
 %rename(_getBindposes) cc::Skeleton::getBindposes;
 %rename(_setBindposes) cc::Skeleton::setBindposes;
 
-%rename(_data) cc::IMemoryImageSource::data;
-%rename(_compressed) cc::IMemoryImageSource::compressed;
+// remove it?
+// %rename(_data) cc::IMemoryImageSource::data;
+// %rename(_compressed) cc::IMemoryImageSource::compressed;
```
Comment (dumganhar) [inline native/tools/swig-config/assets.i]: Yep, remove it.
And add `%ignore cc::IMemoryImageSource` directive.

File: native/tools/swig-config/gfx.i [hunk 2]
```diff
@@ -78,6 +78,14 @@ namespace cc { namespace gfx {
 
 %ignore Device::copyBuffersToTexture;
 %ignore Device::copyTextureToBuffers;
+%ignore Device::createBuffer;
+%ignore Device::createTexture;
```
Comment (dumganhar) [inline native/tools/swig-config/gfx.i:82]: Why are createBuffer & createTexture not needed for binding?

**Comment by dumganhar** [comment]
Also need to remove the elements of `defaultSections` in `tojs/genbindings.py`.

```python
defaultSections = [
    'xr',
    'cocos',
    'video',
    'webview',
    'audio' ,
    'extension',
    'network',
    'gfx',
    'pipeline',
    'render',
    'spine',
    'editor_support',
    'dragonbones',
    'physics',
    'scene',
    'geometry',
    'assets',
    '2d'
]
```

to

```python
defaultSections = [

]

```

File: native/tools/swig-config/assets.i (hunk 1 above)
Comment (mmyduckx) [reply native/tools/swig-config/assets.i:87]: Just sync`skip Device::[Device copyBuffersToTexture copyTextureToBuffers createBuffer createTexture getInstance setOptions getOptions]` in gfx.ini

File: native/tools/swig-config/gfx.i (hunk 2 above)
Comment (novain1102) [reply native/tools/swig-config/gfx.i:82]: > Why are createBuffer & createTexture not needed for binding?

Just sync`skip Device::[Device copyBuffersToTexture copyTextureToBuffers createBuffer createTexture getInstance setOptions getOptions]` in gfx.ini

**Comment by dumganhar** [comment]
Also, there is not a `xr.i` swig configuration for `xr.ini`.

File: native/tools/swig-config/assets.i (hunk 1 above)
Comment (novain1102) [reply native/tools/swig-config/assets.i]: ok

**Review by minggo — APPROVED** [review]

**Review by star-e — APPROVED** [review]

File: native/tools/swig-config/xr.i [hunk 3]
```diff
@@ -6,8 +6,6 @@
 
 // Insert code at the beginning of generated header file (.h)
 %insert(header_file) %{
-// TODO(): need to insert macro_judgement
-// #if (CC_USE_XR > 0)
```
Comment (novain1102) [inline native/tools/swig-config/xr.i]: Already do in cmakelist

**Review by dumganhar — APPROVED** [review]

**Comment by dumganhar** [comment]
@cocos-robot run test cases

**Pull request merged**